### ISSUE TITLE
Looking for translators

### ISSUE THREAD

**Issue by frostworx** [sonic2kk/steamtinkerlaunch#316]
Would be great if someone would fill/complete the orphaned 
- [russian](https://github.com/frostworx/steamtinkerlaunch/blob/master/lang/russian.txt)
- ~~[italian](https://github.com/frostworx/steamtinkerlaunch/blob/master/lang/italian.txt)~~ (**Edit**: was added by @Stefino76)
language files

or even contribute a new language file:

Edit: added a little howto:

You simply need to create a duplicate of [one of the existing language files](https://github.com/frostworx/steamtinkerlaunch/tree/master/lang) _(probably `english.txt`)_ either in the same global directory _(usually `/usr/share/stl/lang/`)_ or in the user directory `~/.config/stl/lang/` and translate every variable in it.
Please make sure to not change the variable names or the opening/closing quotes.
The `XXX`, `YYY`, `ZZZ` strings are placeholders, which are filled by `stl` with variables and should not be removed.

**Comment by jake770321** [comment]
I could add a British English option but it would be mostly spelling changes so I'm not sure if there's any value in adding it. It's really the only way I can contribute to the project though so I'd be happy to

**Comment by frostworx** [comment]
Thank you for the offer, @jake770321. I can not really assess if this would be a benefit for people speaking it.
If you think it is or if this translation option is common in other programs, it would be great to have it as well. :)

**Comment by frostworx** [comment]
Hello @alex37330. That would be great, thank you for the offer!
You're right, I should have mentioned here how to do this:
You simply need to create a duplicate of [one of the existing language files](https://github.com/frostworx/steamtinkerlaunch/tree/master/lang) _(probably english.txt)_ either in the same global directory _(usually `/usr/share/stl/lang/`)_
or in the user directory `~/.config/stl/lang/` and translate every variable in it.
Please make sure to not change the variable names or the opening/closing quotes.
The `XXX`, `YYY`, `ZZZ` strings are placeholders, which are filled by `stl` with variables and should not be removed.

Feel free to ask if anything is unclear.

**Comment by Faalagorn** [comment]
I don't mind someone testing and improving the Polish translation as well :poland: though I try to at least keep it updated.

P.S. I do think separate British and American English would be beneficial. What about hosting the translation files somewhere such as Crowdin, Weblate or Transifex? Some open sourc projects do this:

* Crowdin: https://github.com/EKA2L1/EKA2L1 – https://crowdin.com/project/eka2l1
* Weblate: https://github.com/minetest/minetest  – https://hosted.weblate.org/projects/minetest/minetest/
* Transifex: https://github.com/CleverRaven/Cataclysm-DDA – https://www.transifex.com/cataclysm-dda-translators/cataclysm-dda/

**Comment by frostworx** [comment]
Thanks for the suggestion. I have no clue what the benefit would be, but I do not really want `stl` depend on 3rd party availability if there is no good reason for it.

**Comment by Faalagorn** [comment]
> Thanks for the suggestion. I have no clue what the benefit would be, but I do not really want `stl` depend on 3rd party availability if there is no good reason for it.

These benefit easier translations and translation memory as well as avoid breaking the code by not closing tags, though for a small software project such as stl they might not be as crucial indeed.

**Comment by Stefino76** [comment]
Hello! I'm italian. I coud complete the italian.txt translation. Can I start to translate it and make a Pull Request when I finish the Job?

**Comment by frostworx** [comment]
Hello @Stefino76!
Thank you for the offer! This would be great!
Yeah, feel free to just start whenever you want :)

**Comment by JSvanVeen** [comment]
Hey @frostworx, I'd like to contribute with adding a Dutch translation if that'd be useful. Let me know if it would be!

**Comment by frostworx** [comment]
Hell @JSvanVeen! That would be great! Thanks a lot for the offer! :)
edit: sorry, typo :)

**Comment by frostworx** [comment]
**Thanks a lot to all who offered their help here!**
I'm impressed, surprised and happy that so many people want to help! 😀 👍

**Comment by Stefino76** [comment]
> Hello! I'm italian. I coud complete the italian.txt translation. Can I start to translate it and make a Pull Request when I finish the Job?



> Hey @frostworx, I'd like to contribute with adding a Dutch translation if that'd be useful. Let me know if it would be!

Ok, I'm ready! I've finished my work and I've created a PR in the Master Branch with the file in italian. That's ok?  (This is my second Pull Request in my life, I hope I haven't make mistake!)

**Comment by frostworx** [comment]
Thank you very much @Stefino76! I already merged your Pull Request without any problems. Well done!
The [Changelog](https://github.com/frostworx/steamtinkerlaunch/wiki/Changelog#changes-in-master) is already up to date as well :)

**Comment by frostworx** [comment]
Thank you for the nice feedback @random404-bot and sorry for the late reply - didn't have much time yesterday _(birthday in the family :))_.
Yeah I know that there are several game-specific SpecialK _forks_ and especially FAR seems to be used pretty often, but I'm afraid adding automatic support for those could only work with using a bundled _"database"(csv)_, to find the corresponding download url. Also, it is probably not likely that all those do work under linux _(good to hear FAR does)_.
Take your time with the French translation! Whenever you are ready it is great to have it, also if it would not make it into the next release.

**Comment by frostworx** [comment]
fyi: I added 4 new lines with https://github.com/frostworx/steamtinkerlaunch/commit/762ff0f900f8ed1613e5054c72257df9874aa0c0
Very certain these were the last new ones before the release in a week.
Party on!

**Comment by frostworx** [comment]
Yay, a partial Dutch translation by @JSvanVeen just landed in git master. Thank you very much! 👍

Jjust FYI: I'm going to release the new version tomorrow evening _(AKF from Thursday till ~Sunday)_,
so any translations arriving later will be part of a later _(small)_ release. I hope this is no problem.

If someone has a partial translation and wants it to be part of the release tomorrow, that is fine as well of course.
No hurry and thanks for your time, all! 😀

**Comment by frostworx** [comment]
I just released v7.0: https://github.com/frostworx/steamtinkerlaunch/releases/tag/v7.0
Have much fun with it, and thank you all for your help again! 👍

**Comment by frostworx** [comment]
someone reported in matrix that languages can't be changed. I already have a fix offline, which will be released together with the `stl` → `steamtinkerlaunch` [migration](https://github.com/frostworx/steamtinkerlaunch/issues/350).
I also found some breaking bugs in some languagefiles, which will be fixed with the same commit.
Closing here, as it doesn't seem to attract more translators.
Thanks @all who helped!
